### PR TITLE
Pretty main page

### PR DESCRIPTION
**Pull request by Aliblas**
This branch is built form Mathew's userQueryUpdate branch, so it would be best to merge that first.
I wanted to make recipes resize a to fill their parent div, my efforts failed so more on that later.
Important note: SearchServlet and IdSearchServlet work for testData Recipes, but they can't seem to find Recipes uploaded from the user page.
image

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/NewRecipeServlet.java
Comment (ez-may) [inline src/main/java/com/google/sps/servlets/NewRecipeServlet.java]: Remove this , already on line 60.

File: src/main/java/com/google/sps/servlets/UserSettingsServlet.java
Comment (ez-may) [inline src/main/java/com/google/sps/servlets/UserSettingsServlet.java]: I understand this, but I think you should put this in another class's comments or as a function in a util? Having an entire servlet to hold a comment doesn't seem like the right approach.

**Review by ez-may — CHANGES_REQUESTED** [review]
Change entire servlet for one comment

**Review by ez-may — APPROVED** [review]
LGTM

File: src/main/webapp/components/topbar.jsp
Comment (AlexanderEllis) [inline src/main/webapp/components/topbar.jsp]: Should be `id=`

File: src/main/webapp/javascript/search.js
Comment (AlexanderEllis) [inline src/main/webapp/javascript/search.js:34]: Are this and MainPage.jsp from userQueryUpdate?

**Review by AlexanderEllis — APPROVED** [review]
CSS LGTM (happy to take another look after we merge userQueryUpdate branch)

File: src/main/webapp/javascript/search.js
Comment (ez-may) [reply src/main/webapp/javascript/search.js:34]: I can step in here, but I'm not entirely sure I remember. UserQueyUpdate was more focused on the updates to the servlet and such, but there was a much smaller PR earlier today (Friday 31, July) that was search redirecting

**Pull request merged**
